internal/controllers/user_devices_controller.go

```diff
@@ -277,8 +284,29 @@ func (udc *UserDevicesController) GetUserDevices(c *fiber.Ctx) error {
 	// todo grpc call out to grpc service endpoint in the deviceDefinitionsService udc.deviceDefSvc.GetDeviceDefinitionsByIDs(c.Context(), []string{ "todo"} )
 
 	userID := helpers.GetUserID(c)
+	user, err := udc.usersClient.GetUser(c.Context(), &pb.GetUserRequest{Id: userID})
+	if err != nil {
+		return helpers.ErrorResponseHandler(c, err, fiber.StatusInternalServerError)
+	}
+
+	addr := common.Hex2Bytes(*user.EthereumAddress)
+	// TODO(AE): could reduce db calls with a nested select in UserDevices
```

Comment:
I would like to see a left outer join with `vehicle_nfts`, to see how well that performs.